src/mom5/ocean_param/neutral/ocean_nphysics_util.F90

```diff
@@ -3105,9 +3120,20 @@ subroutine transport_on_rho_gm (Time, Dens, tx_trans_gm, ty_trans_gm)
       ! since the initial value for work is 0.
 
       ! interpolate trans_gm from k-levels to krho-levels
-      do k_rho=1,potrho_nk
-         do k=1,nk-1
+      do k=1,nk-1
+         do j=jsc,jec
+            rho_maxj(j)=maxval(Dens%potrho(isc:iec,j,k+1),mask=Grd%tmask(isc:iec,j,k+1)==1.)
+            rho_minj(j)=minval(Dens%potrho(isc:iec,j,k),mask=Grd%tmask(isc:iec,j,k)==1.)
+         enddo
+         rho_max=maxval(rho_maxj)
+         rho_min=minval(rho_minj)
+         if(rho_max==-huge(rho_max)) exit  ! only rock below this level
+         do k_rho=1,potrho_nk
+            if(rho_max<Dens%potrho_ref(k_rho)) cycle
+            if(rho_min>Dens%potrho_ref(k_rho)) cycle
             do j=jsc,jec
+               if(rho_maxj(j)<Dens%potrho_ref(k_rho)) cycle
+               if(rho_minj(j)>Dens%potrho_ref(k_rho)) cycle
                do i=isc,iec
                   if(     Dens%potrho_ref(k_rho) >  Dens%potrho(i,j,k)  ) then
                       if( Dens%potrho_ref(k_rho) <= Dens%potrho(i,j,k+1)) then 
```

Comment:
Combine to one if statement with an `.and.`?

Reply:
I think you mean a `.or.` if you are talking about the `cycle` lines or do you mean the last two lines. That could be an `.and.`.

Reply:
I meant the last two blocks after the line, yeah.

But don't worry about it.  It's old code and I was just trying to process it in my head.